rfs/src/pack.rs

```diff
@@ -70,12 +72,13 @@ pub async fn pack<P: Into<PathBuf>, S: Store>(
         &writer,
         &mut pool,
         Item(0, root, OsString::from("/"), meta),
+        sender.clone(),
     )
     .await?;
 
     while !list.is_empty() {
         let dir = list.pop_back().unwrap();
-        pack_one(&mut list, &writer, &mut pool, dir).await?;
+        pack_one(&mut list, &writer, &mut pool, dir, sender.clone()).await?;
```

Comment:
Can we pass the sender by reference instead? I think it would be better if you change the code as follows.

```rust
fn pack_one(sender: Option<&Sender<u32>>) {}

pack_one(sender.as_ref()) 
```

the as_ref() will change the Option<T> into Option<&T> so we don't need to copy this everywhere.